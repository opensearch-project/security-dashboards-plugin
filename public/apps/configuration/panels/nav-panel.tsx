/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import { EuiSideNav } from '@elastic/eui';
import { RouteItem } from '../types';

function buildTreeItems(items: RouteItem[]) {
  return [
    {
      name: 'Security',
      id: 'security',
      items: items.map((e: RouteItem) => ({
        id: e.name,
        name: e.name,
        href: '#' + e.href,
        isSelected: window.location.hash.includes(e.href),
      })),
    },
  ];
}

export function NavPanel(props: { items: RouteItem[] }) {
  return <EuiSideNav items={buildTreeItems(props.items)} />;
}
