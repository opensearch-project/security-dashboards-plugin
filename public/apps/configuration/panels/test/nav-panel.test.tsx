/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { shallow } from 'enzyme';
import React from 'react';
import { NavPanel } from '../nav-panel';

describe('Navigation panel', () => {
  it('renders', () => {
    const items = [
      {
        name: 'Item1',
        href: '/item1',
      },
      {
        name: 'Item2',
        href: '/item2',
      },
    ];

    const component = shallow(<NavPanel items={items} />);
    expect(component).toMatchSnapshot();
  });
});
