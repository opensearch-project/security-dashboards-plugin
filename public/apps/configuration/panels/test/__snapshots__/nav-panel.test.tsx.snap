// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation panel renders 1`] = `
<EuiSideNav
  items={
    Array [
      Object {
        "id": "security",
        "items": Array [
          Object {
            "href": "#/item1",
            "id": "Item1",
            "isSelected": false,
            "name": "Item1",
          },
          Object {
            "href": "#/item2",
            "id": "Item2",
            "isSelected": false,
            "name": "Item2",
          },
        ],
        "name": "Security",
      },
    ]
  }
  mobileBreakpoints={
    Array [
      "xs",
      "s",
    ]
  }
/>
`;
